sorc/build_ncep_post.sh

```diff
@@ -8,44 +7,103 @@ SHELL=/bin/sh
 # 01/16 Lin Gan:	Update to use GFS Vertical Structure
 # 07/16 J. Carley:      Generalize for other machines using modules
 # 07/18 Wen Meng:       Set post to v8.0.0 for fv3gfs
+# 10/19 M Kavulich:     Provide machine name as an input argument
 #
 #####################################################################################################
 #####################################################################################################
 
+#List of valid machines:
+validmachines=(theia jet wcoss_dell_p3 wcoss cray-intel hera)
+
+function usage {
+   echo "Usage:"
+   echo " $0 machinename"
+   echo ""
+   echo " Valid values for 'machinename' are: ${validmachines[@]}"
+   exit 1
+}
+
+if [ "$#" -eq 0 ]; then
+   #Check to see if we are building the old way
+   module purge
+   set -x
+   mac=$(hostname | cut -c1-1)
+   mac2=$(hostname | cut -c1-2)
+   if [ $mac2 = tf ] ; then                         # For Theia
+      machine=theia
+   elif [ $mac = f  ] ; then                        # For Jet 
+      machine=jet
+   elif [ $mac = v -o $mac = m  ] ; then            # For Dell
+      machine=wcoss_dell_p3
+   elif [ $mac = t -o $mac = e -o $mac = g ] ; then # For WCOSS
+      machine=wcoss
+   elif [ $mac = l -o $mac = s ] ; then             #    wcoss_c (i.e. luna and surge)
+      export machine=cray-intel
+   elif [ $mac2 = hf ] ; then                       # For Hera
+      machine=hera
+   elif [ $mac = o ] ; then
```

Comment:
Please change low case "o" to upper case "O".

Reply:
@WenMeng-NOAA - Done!